app/models/message.rb

```diff
@@ -5,4 +5,11 @@ class Message < ActiveRecord::Base
 
   validates :body, presence: true
   validates :user_id, presence: true
+
+  def self.create_and_send_email(attrs)
+    message = new(attrs)
+
+    VisitorMailer.one_off_message(message).deliver_later if message.save
+    message
+  end
```

Comment:
I'd create a service object to do just that. I'm not a big fan of coupling the database layer and email sending. But this is simple enough, i'm not too fussed if stays this way.